package/network/utils/comgt/files/ncm.sh

```diff
@@ -117,21 +127,23 @@ proto_ncm_setup() {
 	}
 
 	echo "Starting network $interface"
-	json_get_vars connect
-	eval COMMAND="$connect" gcom -d "$device" -s /etc/gcom/runcommand.gcom || {
-		echo "Failed to connect"
-		proto_notify_error "$interface" CONNECT_FAILED
-		return 1
-	}
-
 	echo "Setting up $ifname"
-	
+
 	proto_init_update "$ifname" 1
 	proto_add_data
 	json_add_string "manufacturer" "$manufacturer"
 	proto_close_data
 	proto_send_update "$interface"
 
+	json_load "$(cat /etc/gcom/ncm.json)"
+	json_select "$manufacturer"
+	json_get_vars connect
+	eval COMMAND="$connect" gcom -d "$device" -s /etc/gcom/runcommand.gcom || {
+		echo "Failed to connect"
+		proto_notify_error "$interface" CONNECT_FAILED
+		return 1
+	}
+
```

Comment:
Dear Matti,
the runcommand.gcom has a control to detect and display the timeout event:
```
 waitfor 15 "OK","ERR","ERROR","COMMAND NOT SUPPORT"
 if % = 0 goto continue
 if % = 1 goto error
 if % = 2 goto error
 if % = 3 goto notsupported

 print "Timeout running AT-command\n"
 exit 1
```
But this is not reported in the logread.
I also enabled the -v (verbose) and the report confirm that no timeout event occurs:

```
Sat Dec 31 20:38:50 2016 daemon.notice netifd: wan3g (1075): sending -> AT+CGDATA="M-MBIM",1,1
Sat Dec 31 20:38:50 2016 daemon.notice netifd: wan3g (1075): comgt 20:38:50 -> @0159 send $env("COMMAND")
Sat Dec 31 20:38:50 2016 daemon.notice netifd: wan3g (1075): comgt 20:38:50 -> @0181 send "^m"
Sat Dec 31 20:38:50 2016 daemon.notice netifd: Network device 'usb0' link is up
Sat Dec 31 20:38:50 2016 daemon.notice netifd: wan3g (1075): comgt 20:38:50 -> @0193 waitfor 15 "OK","ERR","ERROR","COMMAND NOT SUPPORT"
Sat Dec 31 20:38:50 2016 daemon.notice netifd: wan3g (1075):
Sat Dec 31 20:38:50 2016 daemon.notice netifd: wan3g (1075): comgt 20:38:50 -> @0246 if % = 0 goto continue
Sat Dec 31 20:38:50 2016 daemon.notice netifd: wan3g (1075): comgt 20:38:50 -> @0255 goto continue
Sat Dec 31 20:38:50 2016 daemon.notice netifd: wan3g (1075): comgt 20:38:50 -> @0498 :continue
Sat Dec 31 20:38:50 2016 daemon.notice netifd: wan3g (1075): comgt 20:38:50 -> @0509 exit 0
```